src/Video.web.tsx

```diff
@@ -0,0 +1,333 @@
+import React, {
+  forwardRef,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  type RefObject,
+} from 'react';
+import type {VideoRef, ReactVideoProps, VideoMetadata} from './types';
+
+const Video = forwardRef<VideoRef, ReactVideoProps>(
+  (
+    {
+      source,
+      paused,
+      muted,
+      volume,
+      rate,
+      repeat,
+      controls,
+      showNotificationControls = false,
+      poster,
+      onBuffer,
+      onLoad,
+      onProgress,
+      onPlaybackRateChange,
+      onError,
+      onReadyForDisplay,
+      onSeek,
+      onVolumeChange,
+      onEnd,
+      onPlaybackStateChanged,
+    },
+    ref,
+  ) => {
+    const nativeRef = useRef<HTMLVideoElement>(null);
+    const errorHandler = useRef<typeof onError>(onError);
+    errorHandler.current = onError;
+
+    const seek = useCallback(
+      async (time: number, _tolerance?: number) => {
+        if (isNaN(time)) {
+          throw new Error('Specified time is not a number');
+        }
+        if (!nativeRef.current) {
+          console.warn('Video Component is not mounted');
+          return;
+        }
+        time = Math.max(0, Math.min(time, nativeRef.current.duration));
+        nativeRef.current.currentTime = time;
+        onSeek?.({seekTime: time, currentTime: nativeRef.current.currentTime});
+      },
+      [onSeek],
+    );
+
+    const pause = useCallback(() => {
+      if (!nativeRef.current) {
+        return;
+      }
+      nativeRef.current.pause();
+    }, []);
+
+    const resume = useCallback(() => {
+      if (!nativeRef.current) {
+        return;
+      }
+      nativeRef.current.play();
+    }, []);
+
+    const setVolume = useCallback((vol: number) => {
+      if (!nativeRef.current) {
+        return;
+      }
+      nativeRef.current.volume = Math.max(0, Math.min(vol, 100)) / 100;
+    }, []);
+
+    const getCurrentPosition = useCallback(async () => {
+      if (!nativeRef.current) {
+        throw new Error('Video Component is not mounted');
+      }
+      return nativeRef.current.currentTime;
+    }, []);
+
+    const unsupported = useCallback(() => {
+      throw new Error('This is unsupported on the web');
+    }, []);
+
+    useImperativeHandle(
+      ref,
+      () => ({
+        seek,
+        pause,
+        resume,
+        setVolume,
+        getCurrentPosition,
+        // making the video fullscreen does not work with some subtitles polyfils
+        // so I decided to not include it.
```

Comment:
You mean that some subtitles polyfils are not working when video is in fullscreen ? Users can open fullscreen from UI so I think we can implement those and add warning in docs WDYT?

Reply:
Yes, some subtitles polyfill use a canvas or other HTML elements to draw on top of the video, and those do not work if we simply full-screen the video.

I was thinking of implementing the full-screen feature after adding the polyfill, since it's possible to fullscreen a container of both the video and the polyfill.

Reply:
I understand, but can we for now implement "built-in" fullscreen? We can later edit it to support subtitles polyfills

Reply:
Kay, i'll add that

Reply:
Thanks!

Reply:
Done, I put only the video element on fullscreen tho so controls are not shown in fullscreen.

On my website, I put the whole body in fullscreen, but that might not be doable for some apps. Do you have an idea how to improve that?

Reply:
I was thinking about using this https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullscreen

Reply:
Yes, that's what I used. Controls can't be shown using this tho.

Reply:
https://github.com/TheWidlarzGroup/react-native-video/assets/83216158/2f45c2b2-5dea-4dc6-85fd-5805d9ba0fd7

I have checked it and it's working

Reply:
Built-in (HTML5 defaults) controls will show but not custom ones

Reply:
I see three ways of fixing this:
- have a props for a ref of the item that needs to be in fullscreen (could be a ref to document.body)
- have controls as child of the Video element
- add an html id/class `player-controls` and we could find the closest parent with this class to fullscreen

The second one is the most "react-like" but it kinda breaks expectation of having a simple Video element + it's a breaking.

The other two feels more like escape hatch.

What does the fullscreen behavior does on android/ios? I must say I never used it and always used fullscreen on the whole route via the router.

Reply:
Fullscreen on iOS/Android also doesn't allow custom controls so it's not a problem (maybe in future we will add it)